### 1. Connection test

<br><br>
<b>$ conda install pywin32</b>

In [2]:
import win32com.client

instCpCybos = win32com.client.Dispatch("CpUtil.CpCybos")
print(instCpCybos.IsConnect)

1


<br><br>
<b>1 if connected to CYBOS PLus  
else check for Admin Access</b>

---
### 2. StockCode 

<br><br>
<b> module CpUtil has CpStockCode Class w/ severall methods to access StockCode</b>

In [4]:
instCpStockCode = win32com.client.Dispatch("CpUtil.CpStockCode")
instCpStockCode.GetCount()

3410

<br><br>
<b>GetCount() Method returns num of Stocks in the market</b>

<b>There are 3410 stocks in the market w/ indexes 0 to 3409</b>

<b>Use GetData method to map index -> Stock Name</b>

In [6]:
instCpStockCode = win32com.client.Dispatch("CpUtil.CpStockCode")
instCpStockCode.GetData(1, 0)

'동화약품'

<b>GetData(a, b)  
+ a=0 : index -> StockCode
+ a=1 : index -> StockName
+ a=2 : index -> FullCode  
+ b = index num
</b>

In [7]:
instCpStockCode.GetData(0, 0)

'A000020'

<br><br>
<b>Mapping Stock Name to Stock Code</b>

<b>Use NameToCode() Method  
To get the exact Stockname [here](http://data.krx.co.kr/)</b>

In [13]:
naverCode = instCpStockCode.NameToCode('NAVER')
naverIndex = instCpStockCode.CodeToIndex(naverCode)

In [14]:
naverCode

'A035420'

In [15]:
naverIndex

873

---
### 3. CpCodeMgr

<br><br>
<b>CpCodeMgr Class probides severall methods to get code info and list</b>

In [16]:
instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")

<br><br>
<b>GetStockListByMarket(1) returns Stock list as Tuple</b>

In [17]:
codeList = instCpCodeMgr.GetStockListByMarket(1)
codeList

('A000020',
 'A000040',
 'A000050',
 'A000060',
 'A000070',
 'A000075',
 'A000080',
 'A000087',
 'A000100',
 'A000105',
 'A000120',
 'A000140',
 'A000145',
 'A000150',
 'A000155',
 'A000157',
 'A000180',
 'A000210',
 'A000215',
 'A000220',
 'A000225',
 'A000227',
 'A000230',
 'A000240',
 'A000270',
 'A000300',
 'A000320',
 'A000325',
 'A000370',
 'A000390',
 'A000400',
 'A000430',
 'A000480',
 'A000490',
 'A000500',
 'A000520',
 'A000540',
 'A000545',
 'A000547',
 'A000590',
 'A000640',
 'A000650',
 'A000660',
 'A000670',
 'A000680',
 'A000700',
 'A000720',
 'A000725',
 'A000760',
 'A000810',
 'A000815',
 'A000850',
 'A000860',
 'A000880',
 'A000885',
 'A00088K',
 'A000890',
 'A000910',
 'A000950',
 'A000970',
 'A000990',
 'A000995',
 'A001020',
 'A001040',
 'A001045',
 'A00104K',
 'A001060',
 'A001065',
 'A001067',
 'A001070',
 'A001080',
 'A001120',
 'A001130',
 'A001140',
 'A001200',
 'A001210',
 'A001230',
 'A001250',
 'A001260',
 'A001270',
 'A001275',
 'A001290',
 'A001340',
 'A0

<br><br>
<b>This Tuple is not usefull, nobody wants to remember Stockcodes  
In this case, dictionary type might be useful </b>

In [20]:
instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")
codeList = instCpCodeMgr.GetStockListByMarket(1)

kospi = {code: instCpCodeMgr.CodeToName(code) for code in codeList}

kospi

{'A000020': '동화약품',
 'A000040': 'KR모터스',
 'A000050': '경방',
 'A000060': '메리츠화재',
 'A000070': '삼양홀딩스',
 'A000075': '삼양홀딩스우',
 'A000080': '하이트진로',
 'A000087': '하이트진로2우B',
 'A000100': '유한양행',
 'A000105': '유한양행우',
 'A000120': 'CJ대한통운',
 'A000140': '하이트진로홀딩스',
 'A000145': '하이트진로홀딩스우',
 'A000150': '두산',
 'A000155': '두산우',
 'A000157': '두산2우B',
 'A000180': '성창기업지주',
 'A000210': 'DL',
 'A000215': 'DL우',
 'A000220': '유유제약',
 'A000225': '유유제약1우',
 'A000227': '유유제약2우B',
 'A000230': '일동홀딩스',
 'A000240': '한국앤컴퍼니',
 'A000270': '기아',
 'A000300': '대유플러스',
 'A000320': '노루홀딩스',
 'A000325': '노루홀딩스우',
 'A000370': '한화손해보험',
 'A000390': '삼화페인트',
 'A000400': '롯데손해보험',
 'A000430': '대원강업',
 'A000480': '조선내화',
 'A000490': '대동',
 'A000500': '가온전선',
 'A000520': '삼일제약',
 'A000540': '흥국화재',
 'A000545': '흥국화재우',
 'A000547': '흥국화재2우B',
 'A000590': 'CS홀딩스',
 'A000640': '동아쏘시오홀딩스',
 'A000650': '천일고속',
 'A000660': 'SK하이닉스',
 'A000670': '영풍',
 'A000680': 'LS네트웍스',
 'A000700': '유수홀딩스',
 'A000720': '현대건설',
 'A000725': '현대건설우

<br><br>
<b>or you can store it in the .csv file</b>

In [22]:
import os
import win32com.client

instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")
codeList = instCpCodeMgr.GetStockListByMarket(1)

kospi = {code: instCpCodeMgr.CodeToName(code) for code in codeList}

fin = open(os.path.join(os.getcwd(), 'kospi.csv'), 'w', encoding = 'UTF-8')
for key, value in kospi.items():
           fin.write(f"{key}, {value}\n")
fin.close()

<br><br>
<b>Code list from the GetStockListByMartket method contains ETF and ETN info  
w/ GetStockSectionKind method we can check whether the stock is etf or etn</b>

<br><br>
<b>GetStockSectionKind method returns 1 for stock, 10 for ETF, 17 for ETN  
for some detail, check [here](https://money2.daishin.com/e5/mboard/ptype_basic/HTS_Plus_Helper/DW_Basic_Read_Page.aspx?boardseq=284&seq=11&page=1&searchString=Mgr&p=&v=&m=)</b>

In [23]:
instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")
codeList = instCpCodeMgr.GetStockListByMarket(1)

for i, code in enumerate(codeList):
    stockKind = instCpCodeMgr.GetStockSectionKind(code)
    name = instCpCodeMgr.CodeToName(code)
    print(i, code, stockKind, name)

0 A000020 1 동화약품
1 A000040 1 KR모터스
2 A000050 1 경방
3 A000060 1 메리츠화재
4 A000070 1 삼양홀딩스
5 A000075 1 삼양홀딩스우
6 A000080 1 하이트진로
7 A000087 1 하이트진로2우B
8 A000100 1 유한양행
9 A000105 1 유한양행우
10 A000120 1 CJ대한통운
11 A000140 1 하이트진로홀딩스
12 A000145 1 하이트진로홀딩스우
13 A000150 1 두산
14 A000155 1 두산우
15 A000157 1 두산2우B
16 A000180 1 성창기업지주
17 A000210 1 DL
18 A000215 1 DL우
19 A000220 1 유유제약
20 A000225 1 유유제약1우
21 A000227 1 유유제약2우B
22 A000230 1 일동홀딩스
23 A000240 1 한국앤컴퍼니
24 A000270 1 기아
25 A000300 1 대유플러스
26 A000320 1 노루홀딩스
27 A000325 1 노루홀딩스우
28 A000370 1 한화손해보험
29 A000390 1 삼화페인트
30 A000400 1 롯데손해보험
31 A000430 1 대원강업
32 A000480 1 조선내화
33 A000490 1 대동
34 A000500 1 가온전선
35 A000520 1 삼일제약
36 A000540 1 흥국화재
37 A000545 1 흥국화재우
38 A000547 1 흥국화재2우B
39 A000590 1 CS홀딩스
40 A000640 1 동아쏘시오홀딩스
41 A000650 1 천일고속
42 A000660 1 SK하이닉스
43 A000670 1 영풍
44 A000680 1 LS네트웍스
45 A000700 1 유수홀딩스
46 A000720 1 현대건설
47 A000725 1 현대건설우
48 A000760 1 이화산업
49 A000810 1 삼성화재
50 A000815 1 삼성화재우
51 A000850 1 화천기공
52 A000860 1 강남제비스코
53 A000880

474 A013000 1 세우글로벌
475 A013360 1 일성건설
476 A013520 1 화승코퍼레이션
477 A013570 1 디와이
478 A013580 1 계룡건설
479 A013700 1 까뮤이앤씨
480 A013870 1 지엠비코리아
481 A013890 1 지누스
482 A014130 1 한익스프레스
483 A014160 1 대영포장
484 A014280 1 금강공업
485 A014285 1 금강공업우
486 A014440 1 영보화학
487 A014530 1 극동유화
488 A014580 1 태경비케이
489 A014680 1 한솔케미칼
490 A014710 1 사조씨푸드
491 A014790 1 한라
492 A014820 1 동원시스템즈
493 A014825 1 동원시스템즈우
494 A014830 1 유니드
495 A014910 1 성문전자
496 A014915 1 성문전자우
497 A014990 1 인디에프
498 A015020 1 이스타코
499 A015230 1 대창단조
500 A015260 1 에이엔피
501 A015350 1 부산가스
502 A015360 1 예스코홀딩스
503 A015540 1 쎌마테라퓨틱스
504 A015590 1 큐로
505 A015760 1 한국전력
506 A015860 1 일진홀딩스
507 A015890 1 태경산업
508 A016090 1 대현
509 A016360 1 삼성증권
510 A016380 1 KG동부제철
511 A016385 1 KG동부제철우
512 A016450 1 한세예스24홀딩스
513 A016580 1 환인제약
514 A016590 1 신대양제지
515 A016610 1 DB금융투자
516 A016710 1 대성홀딩스
517 A016740 1 두올
518 A016800 1 퍼시스
519 A016880 1 웅진
520 A017040 1 광명전기
521 A017180 1 명문제약
522 A017370 1 우신시스템
523 A017390 1 서울가스
524 A017550 1 수산중공업
525 

1259 A316300 10 KINDEX 싱가포르리츠
1260 A316670 10 KOSEF 코스닥150
1261 A317400 1 자이에스앤디
1262 A319640 10 TIGER 골드선물(H)
1263 A319870 10 KBSTAR KRX300미국달러선물혼합
1264 A321410 10 KODEX 멀티에셋하이인컴(H)
1265 A322000 1 현대에너지솔루션
1266 A322120 10 KINDEX 스마트퀄리티
1267 A322130 10 KINDEX 스마트로우볼
1268 A322150 10 KINDEX 스마트하이베타
1269 A322400 10 HANARO e커머스
1270 A322410 10 HANARO 고배당
1271 A323410 1 카카오뱅크
1272 A325010 10 KODEX Fn성장
1273 A325020 10 KODEX 배당가치
1274 A326030 1 SK바이오팜
1275 A326230 10 KBSTAR 내수주플러스
1276 A326240 10 KBSTAR IT플러스
1277 A328370 10 ARIRANG 코스피TR
1278 A329200 10 TIGER 부동산인프라고배당
1279 A329650 10 KODEX TRF3070
1280 A329660 10 KODEX TRF5050
1281 A329670 10 KODEX TRF7030
1282 A329750 10 TIGER 미국달러단기채권액티브
1283 A330590 3 롯데리츠
1284 A331910 10 KOSEF Fn중소형
1285 A332500 10 KINDEX 200TR
1286 A332610 10 ARIRANG 미국단기우량회사채
1287 A332620 10 ARIRANG 미국장기우량회사채
1288 A332930 10 HANARO 200TR
1289 A332940 10 HANARO MSCI Korea TR
1290 A333940 10 ARIRANG KS로우볼가중TR
1291 A333950 10 ARIRANG KS로우사이즈가중TR
1292 A333960 10 ARIRANG 

1620 Q610004 17 메리츠 미국레버리지 인플레이션국채ETN(H)
1621 Q610005 17 메리츠 레버리지 KRX300 선물 ETN
1622 Q610006 17 메리츠 인버스 2X KRX300 선물 ETN
1623 Q700001 17 하나 코스피 변동성추세 추종 양매도 ETN
1624 Q700002 17 하나 KRX300 ETN
1625 Q700003 17 하나 KRX BBIG K-뉴딜 ETN


<br><br>


---
### 4. Past Data

<br><br>

<b>StockChart Class from CpSysDib Module provides past data of a stock</b>

<b>Let's create a instance first</b>

In [25]:
import win32com.client
instStockChart = win32com.client.Dispatch("CpSysDib.StockChart")

<b>StockChart Class uses Request/Reply Protocol  
Use SetInputValue method to request</b>

<b>object.SetInputValue(type, value)  
Type: Value
+ 0: StockCode(String)   
    
+ 1: Request type(char)  
    '1' for period
    '2' for num
+ 2: Request EndTime(ulong)      
    YYYYMMDD, Default(0) Recent exchange
+ 3: Request StartTime(ulong)   
    YYYYMMDD
+ 4: Request number(ulong)    
    num of request data
+ 5: Field(long array)     
    date, time, price, ... 
+ 6: Chart Type(Char)     
    'D' for day
    'T' for tick ...
  
    for detailed info, check [here](https://money2.daishin.com/e5/mboard/ptype_basic/HTS_Plus_Helper/DW_Basic_Read_Page.aspx?boardseq=284&seq=102&page=1&searchString=CpSysDib.StockChart&p=&v=&m=)
</b>

In [29]:
instStockChart.SetInputValue(0, "A003540")
instStockChart.SetInputValue(1, ord('2'))
instStockChart.SetInputValue(4, 10)
instStockChart.SetInputValue(5, 5)
instStockChart.SetInputValue(6, ord('D'))
instStockChart.SetInputValue(9, ord('1'))

<br><br>
<b>Code above implies
+ 0: data about "A003540" ('대신증권')
+ 1: number of data
+ 4: 10 number of data
+ 5: 10 number of close price
+ 6: 10 number of close price of the day
+ 9: 10 numver of 'adjusted' close price of the day  
</b>

<br><br>
<b>After initial setting, we can request the data </b>

In [31]:
instStockChart.BlockRequest()

0

<b>when you request a data, server reploes  
Use GetHeaderValue and GetDataValue method to get data</b>

In [32]:
numData = instStockChart.GetHeaderValue(3)

In [33]:
numData

10

<b>number of data is valid, now print the close price of the 10 days</b>

In [34]:
for i in range(numData):
    print(instStockChart.GetDataValue(0, i))

18650
18400
18200
18100
18100
18100
18150
17850
17850
17900


<b>We only requested 1 field, close price of the stock.  
so set the first argument of the method as 0,  <br><br>
if you requested more than one field,  
first argument will be index of the field
</b>

In [35]:
import win32com.client
instStockChart = win32com.client.Dispatch("CpSysDib.StockChart")

In [36]:
instStockChart.SetInputValue(0, "A003540")
instStockChart.SetInputValue(1, ord('2'))
instStockChart.SetInputValue(4, 10)
instStockChart.SetInputValue(5, (0, 2, 3, 4, 5, 8))
instStockChart.SetInputValue(6, ord('D'))
instStockChart.SetInputValue(9, ord('1'))

<br><br>
<b>This time we request 6 fileds,  
Date, Open, High, Low, Close Price, Tradaing Volume</b>

In [37]:
instStockChart.BlockRequest()

0

In [38]:
numData = instStockChart.GetHeaderValue(3)
numField = instStockChart.GetHeaderValue(1)

In [39]:
numData

10

In [40]:
numField

6

In [43]:
print(f"{'Date':>8} {'Open':>5} {'High':>5} {'Close':>5} {'Low':>5} {'Volume':>6}")
for i in range(numData):
    for j in range(numField):
        print(instStockChart.GetDataValue(j, i), end=" ")
    print("")

    Date  Open  High Close   Low Volume
20210812 18550 18700 18500 18650 91651 
20210811 18200 18750 18200 18400 220119 
20210810 18150 18350 18150 18200 67904 
20210809 18200 18300 17900 18100 113420 
20210806 18300 18300 18100 18100 30238 
20210805 18200 18250 18050 18100 59636 
20210804 17850 18150 17800 18150 51884 
20210803 17950 18050 17750 17850 79920 
20210802 17800 17900 17550 17850 93766 
20210730 18350 18400 17850 17900 203298 


<br><br>
<b>This time we will request data from ceratin period</b>

In [44]:
import win32com.client
instStockChart = win32com.client.Dispatch("CpSysDib.StockChart")

In [45]:
instStockChart.SetInputValue(0, "A003540")
instStockChart.SetInputValue(1, ord('1'))
instStockChart.SetInputValue(2, 20161031)
instStockChart.SetInputValue(3, 20161020)
instStockChart.SetInputValue(5, (0, 2, 3, 4, 5, 8))
instStockChart.SetInputValue(6, ord('D'))
instStockChart.SetInputValue(9, ord('1'))

In [46]:
instStockChart.BlockRequest()

0

In [47]:
numData = instStockChart.GetHeaderValue(3)
numField = instStockChart.GetHeaderValue(1)

In [48]:
numData

8

In [49]:
numField

6

In [50]:
print(f"{'Date':>8} {'Open':>5} {'High':>5} {'Close':>5} {'Low':>5} {'Volume':>6}")
for i in range(numData):
    for j in range(numField):
        print(instStockChart.GetDataValue(j, i), end=" ")
    print("")

    Date  Open  High Close   Low Volume
20161031 10600 10650 10450 10550 141131 
20161028 10750 10800 10550 10650 62901 
20161027 10650 10750 10550 10700 77954 
20161026 10950 10950 10550 10600 173509 
20161025 10900 10950 10700 10850 160825 
20161024 10850 10950 10800 10850 80856 
20161021 10800 10900 10700 10850 209439 
20161020 10750 10900 10650 10850 137679 


<br><br>

---
### 5. PER, EPS

<br><br>
<b>MarketEye Class from CpSysDib Module provides indexes such as PER amd EPS</b>

In [52]:
import win32com.client

instMarketEye = win32com.client.Dispatch("CpSysDib.MarketEye")

In [53]:
instMarketEye.SetInputValue(0, (4, 67, 70, 111))
instMarketEye.SetInputValue(1, 'A003540')

<br><br>
<b>SetInputValue() method to set field value  
4 for market Price, 67 for PER, 70 for EPS, Recent Q  
for detail [here](https://money2.daishin.com/e5/mboard/ptype_basic/HTS_Plus_Helper/DW_Basic_Read_Page.aspx?boardseq=284&seq=131&page=1&searchString=MarketEye&p=&v=&m=)<b>

In [54]:
instMarketEye.BlockRequest()

0

<b>GetDataValue(filed, stock) method to get data</b>

In [56]:
print("현재가: ", instMarketEye.GetDataValue(0, 0))
print("PER: ", instMarketEye.GetDataValue(1, 0))
print("EPS: ", instMarketEye.GetDataValue(2, 0))
print("최근분기년월: ", instMarketEye.GetDataValue(3, 0))

현재가:  18650
PER:  8.15999984741211
EPS:  2283
최근분기년월:  202103


<br><br>

---

### 6. Setting Algorithm

<br><br>
<b>Let's set up a simple trading algorithm for excercise</b>

<b>Our first alorithm will find  
+ 1) Massive Trading Volume : increasing more than 1,000%
+ 2) Low PBR : PBR \<4 at Massive Trading </b>

<br><br>
<b>To get the Trading Volume, we can use StockChart Class  
below is a code saving recent 60 days' trading volume of certain stock </b>

In [57]:
import win32com.client
instStockChart = win32com.client.Dispatch("CpSysDib.StockChart")

In [58]:
instStockChart.SetInputValue(0, "A003540")
instStockChart.SetInputValue(1, ord('2'))
instStockChart.SetInputValue(4, 60)
instStockChart.SetInputValue(5, 8)
instStockChart.SetInputValue(6, ord('D'))
instStockChart.SetInputValue(9, ord('1'))

In [59]:
instStockChart.BlockRequest()


0

In [61]:
numdata = instStockChart.GetHeaderValue(3)
volumes = [instStockChart.GetDataValue(0, i) for i in range(numdata)]

In [62]:
volumes

[91651,
 220119,
 67904,
 113420,
 30238,
 59636,
 51884,
 79920,
 93766,
 203298,
 59591,
 117991,
 55537,
 81242,
 40213,
 63654,
 73452,
 113086,
 68925,
 71363,
 93696,
 59451,
 136122,
 54568,
 239750,
 102254,
 132708,
 49583,
 72523,
 152929,
 118447,
 146146,
 226576,
 576868,
 109951,
 109418,
 152176,
 68906,
 150054,
 115030,
 129447,
 180004,
 190456,
 192108,
 100672,
 132130,
 104614,
 80582,
 111259,
 169128,
 160145,
 96570,
 138710,
 124179,
 141297,
 115024,
 228485,
 158690,
 121123,
 161480]

<br><br>
<b>Now we wil compare recent trading volume w/ 59 days' average  
volumes[0] will ve our recent trading volume</b>

In [63]:
averageVolume = (sum(volumes) - volumes[0]) / (len(volumes) - 1)

In [64]:
if volumes[0] > averageVolume * 10:
    print("Condition: True")
else:
    print("Condition: False, ", volumes[0] / averageVolume)

Condition: False,  0.7338549864572128


<b>now let's generalize the program  
    check the condition for all the stocks in the market</b>

<b>first_algorithm.py</b>

<br><br>


---
### 7. Comparing PER in the Sector

<br><br>
<b>PER is a great index for investment.  
However, PER can differ from industry sector or company size.</b>

<br><br>
<b>Let's compare the company's PER w/ its industry's average   
by using GetIndustryList method</b>

In [65]:
import win32com.client

In [66]:
instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")

In [67]:
industryCodeList = instCpCodeMgr.GetIndustryList()

In [69]:
for industryCode in industryCodeList:
    print(industryCode, instCpCodeMgr.GetIndustryName(industryCode))

001 종합주가지수
002 대형(시가총액)
003 중형(시가총액)
004 소형(시가총액)
005 음식료품
006 섬유,의복
007 종이,목재
008 화학
009 의약품
010 비금속광물
011 철강,금속
012 기계
013 전기,전자
014 의료정밀
015 운송장비
016 유통업
017 전기가스업
018 건설업
019 운수창고
020 통신업
021 금융업
022 은행
024 증권
025 보험
026 서비스업
027 제조업
130 KOSPI200 경기방어소비재
131 KOSPI200 중소형주지수
132 KOSPI200 산업재
133 KOSPI200 헬스케어
141 KOSPI200 건설
142 KOSPI200 중공업
143 KOSPI200 철강/소재
144 KOSPI200 에너지/화학
145 KOSPI200 정보기술
146 KOSPI200 금융
147 KOSPI200 생활소비재
148 KOSPI200 경기소비재
149 코스피 200 에너지화학 레버리지지수
150 미국달러선물 레버리지지수
151 미국달러선물지수
152 미국달러선물인버스지수
153 코스피 200 커버드콜 5% OTM
154 코스피 200 프로텍티브풋 OTM 5%
155 KOSPI200 동일가중지수
156 KOSPI100 동일가중지수
157 KOSPI50 동일가중지수
165 코스피 200 정보기술 레버리지지수
166 코스피 200 금융 레버리지지수
167 코스피 200 경기소비재 레버리지지수
170 KTOP 30
177 KOSPI200레버리지지수
178 F-KOSPI200인버스지수
179 KOSPI200선물지수
180 KOSPI200지수
181 KOSPI100지수
182 KOSPI50지수
184 코스피 200 TR
185 코스피 200 NTR
186 KRX-IHS Markit 코스피 200 예측 고배당 30
187 KRX-IHS Markit 코스피 200 예측 배당성장 30
190 코스피 200 고배당지수
191 코스피 200 저변동성지수
192 코스피 고배당 50지수
193 코스피 배당성장 50지수


<br><br>
<b>Let's say we are interested in the Food Industry</b>

In [71]:
instCpCodeMgr = win32com.client.Dispatch("CpUtil.CpCodeMgr")
tarketCodeList = instCpCodeMgr.GetGroupCodeList(5) # 5 for '음식료품'

for code in tarketCodeList:
    print(code, instCpCodeMgr.CodeToName(code))

A011150 CJ씨푸드
A011155 CJ씨푸드1우
A097950 CJ제일제당
A097955 CJ제일제당 우
A023150 MH에탄올
A005610 SPC삼립
A002140 고려산업
A003920 남양유업
A003925 남양유업우
A004370 농심
A001680 대상
A001685 대상우
A001790 대한제당
A001795 대한제당우
A001130 대한제분
A049770 동원F&B
A280360 롯데제과
A005300 롯데칠성
A005305 롯데칠성우
A002270 롯데푸드
A027740 마니커
A033920 무학
A000890 보해양조
A005180 빙그레
A003960 사조대림
A008040 사조동아원
A014710 사조씨푸드
A006090 사조오양
A145990 삼양사
A145995 삼양사우
A003230 삼양식품
A248170 샘표식품
A004410 서울식품
A004415 서울식품우
A136490 선진
A007310 오뚜기
A271560 오리온
A006980 우성사료
A002600 조흥
A264900 크라운제과
A26490K 크라운제과우
A036580 팜스코
A017810 풀무원
A000080 하이트진로
A000087 하이트진로2우B
A003680 한성기업
A101530 해태제과식품


<br><br>
<b>Let's load average PER of the food industry, w/ MarketEye Class</b>

In [72]:
instMarketEye = win32com.client.Dispatch("CpSysDib.MarketEye")

In [73]:
# Get PER
instMarketEye.SetInputValue(0, 67)
instMarketEye.SetInputValue(1, tarketCodeList)

In [74]:
instMarketEye.BlockRequest()

0

In [75]:
numStock = instMarketEye.GetHeaderValue(2)

In [76]:
sumPER = sum((instMarketEye.GetDataValue(0, i) for i in range(numStock)))

In [77]:
print("Average PER of Food Industry: ", sumPER / numStock)

Average PER of Food Industry:  13.995531939445659


<br><br>

---
### 8. Buying / Selling Stocks

<br><br>

<b>...Writing</b>